File: golang/version3.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
)

// Version 3: Convert number to words in Spanish using native Go library
// Usage: go run version3.go <number>
// Example: go run version3.go 10

func main() {
	number := "10"
	if len(os.Args) > 1 {
		number = os.Args[1]
	}

	num, err := strconv.ParseInt(number, 10, 64)
	if err != nil {
		fmt.Printf("Error: %v\n", err)
		return
	}

	result := numberToSpanishWords(num)
	fmt.Println(result)
}

func numberToSpanishWords(n int64) string {
	if n == 0 {
		return "cero"
	}

	units := []string{"", "uno", "dos", "tres", "cuatro", "cinco", "seis", "siete", "ocho", "nueve"}
	teens := []string{"diez", "once", "doce", "trece", "catorce", "quince", "dieciséis", "diecisiete", "dieciocho", "diecinueve"}
	tens := []string{"", "", "veinte", "treinta", "cuarenta", "cincuenta", "sesenta", "setenta", "ochenta", "noventa"}
	hundreds := []string{"", "ciento", "doscientos", "trescientos", "cuatrocientos", "quinientos", "seiscientos", "setecientos", "ochocientos", "novecientos"}

	if n < 10 {
		return units[n]
	}
	if n < 20 {
		return teens[n-10]
	}
	if n < 100 {
		t := n / 10
		u := n % 10
		if u == 0 {
			return tens[t]
		}
		return tens[t] + " y " + units[u]
	}
	if n < 1000 {
		h := n / 100
		rest := n % 100
		if h == 1 && rest == 0 {
			return "cien"
		}
		if rest == 0 {
			return hundreds[h]
		}
		return hundreds[h] + " " + numberToSpanishWords(rest)
	}
	if n < 1000000 {
		thousands := n / 1000
		rest := n % 1000
		if thousands == 1 {
			if rest == 0 {
				return "mil"
			}
			return "mil " + numberToSpanishWords(rest)
		}
		if rest == 0 {
			return numberToSpanishWords(thousands) + " mil"
		}
		return numberToSpanishWords(thousands) + " mil " + numberToSpanishWords(rest)
	}

	return strconv.FormatInt(n, 10)
}

File: rust/version3.rs
use std::env;

// Version 3: Convert number to words in Spanish using native Rust library
// Usage: cargo run --bin version3 -- <number>
// Example: cargo run --bin version3 -- 10

fn number_to_spanish_words(n: i64) -> String {
    if n == 0 {
        return "cero".to_string();
    }
    
    let units = ["", "uno", "dos", "tres", "cuatro", "cinco", "seis", "siete", "ocho", "nueve"];
    let teens = ["diez", "once", "doce", "trece", "catorce", "quince", "dieciséis", "diecisiete", "dieciocho", "diecinueve"];
    let tens = ["", "", "veinte", "treinta", "cuarenta", "cincuenta", "sesenta", "setenta", "ochenta", "noventa"];
    let hundreds = ["", "ciento", "doscientos", "trescientos", "cuatrocientos", "quinientos", "seiscientos", "setecientos", "ochocientos", "novecientos"];
    
    if n < 10 {
        return units[n as usize].to_string();
    }
    if n < 20 {
        return teens[(n - 10) as usize].to_string();
    }
    if n < 100 {
        let t = (n / 10) as usize;
        let u = (n % 10) as usize;
        if u == 0 {
            return tens[t].to_string();
        }
        return format!("{} y {}", tens[t], units[u]);
    }
    if n < 1000 {
        let h = (n / 100) as usize;
        let rest = n % 100;
        if h == 1 && rest == 0 {
            return "cien".to_string();
        }
        if rest == 0 {
            return hundreds[h].to_string();
        }
        return format!("{} {}", hundreds[h], number_to_spanish_words(rest));
    }
    if n < 1000000 {
        let thousands = n / 1000;
        let rest = n % 1000;
        if thousands == 1 {
            if rest == 0 {
                return "mil".to_string();
            }
            return format!("mil {}", number_to_spanish_words(rest));
        }
        if rest == 0 {
            return format!("{} mil", number_to_spanish_words(thousands));
        }
        return format!("{} mil {}", number_to_spanish_words(thousands), number_to_spanish_words(rest));
    }
    
    n.to_string()
}

fn main() {
    let args: Vec<String> = env::args().collect();
    let number = args.get(1).unwrap_or(&"10".to_string()).clone();
    
    match number.parse::<i64>() {
        Ok(num) => {
            let result = number_to_spanish_words(num);
            println!("{}", result);
        }
        Err(e) => {
            eprintln!("Error: {}", e);
        }
    }
}
